pkgs/misc/cups/drivers/brother/dcpj315w/default.nix

```diff
@@ -0,0 +1,102 @@
+{ lib
+, stdenv
+, fetchurl
+, cups
+, dpkg
+, gnused
+, makeWrapper
+, ghostscript
+, file
+, a2ps
+, coreutils
+, gnugrep
+, which
+, gawk
+}:
+
+let
+  version = "1.1.3";
+  model = "dcpj315w";
+in
+rec {
+  driver = stdenv.mkDerivation {
+    pname = "${model}-lpr";
+    inherit version;
+
+    src = fetchurl {
+      url= "https://download.brother.com/welcome/dlf005591/dcpj315wlpr-${version}-1.i386.deb";
+      sha256 = "1myc6jrbkk6rbx1ynhac24l3hlkz7dmic557rz6q69fc54m69zmi";
+    };
+
+    nativeBuildInputs = [ dpkg makeWrapper ];
+    buildInputs = [ cups ghostscript a2ps gawk ];
+    unpackPhase = "dpkg-deb -x $src $out";
+
+    installPhase = ''
+      substituteInPlace $out/opt/brother/Printers/${model}/lpd/filter${model} \
+      --replace /opt "$out/opt"
+
+      substituteInPlace $out/opt/brother/Printers/${model}/inf/brdcpj315wrc \
+      --replace "Letter" "A4"
```

Comment:
is this broken or a nix specific fix, or you prefer the default for something is A4?

Reply:
If I remember well, the printer was always printing letter formarts even if A4 was selected. This lead to a shift of the printed area on the A4 page. As I only print A4, this did the trick for me. However, I don't know whether a print on letter would work fine with this setup.

Reply:
I am sympathetic to the fact that you need this for your use of the driver, but I am concerned about changing the default behaviour if it only fixes a subset of users (and breaks the rest).

if you can patch the bug, that would be optimal, or at least submit that upstream, assuming brother takes feedback.

otherwise, we should probably expose an attr for controlling this so that we can have a mirror package or you can override things yourself:
```nix
# dcpj315w/default.nix
{ size ? "Letter"
, lib
```
```nix
# all-packages.nix
dcpj315w-lpr = callPackage ../path/to/dcpj315w/default.nix {}; 
dcpj315w-a4-lpr = callPackage ../path/to/dcpj315w/default.nix { size = "A4" }; 
```
or you can do something like:
```nix
# configuration.nix
dcpj315w-lpr.override { size = "A4"; }
```

Reply:
I don't know whether fixing this upstream is realistic. Actually, I cannot even test this, as I am lacking letter format paper. It would be great, if someone from the community could verify the behavior I observed for A4, and also check what happens when you actually print on letter. However, I don't know who else owns a dcpj 315

For me overriding seams plausible. If the driver is installed by others they might open up issues and then these could be investigated as a community effort. What do you think?

Reply:
yeah, we likely need more users with hardware to confirm anything for sure. I will defer to your judgement on what packages to expose, but I would be keen to keep a "default version" of this driver around, even if you are not using it. (feel free to fix things up and I will throw you a review.)